src/frontend/query_commands.ml

```diff
@@ -502,6 +502,47 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
     if path = "" then `Invalid_context else
       Locate.get_doc ~config
         ~env ~local_defs ~comments ~pos (`User_input path)
+  | Syntax_document (patho, pos) ->
+    let typer = Mpipeline.typer_result pipeline in
+    let pos = Mpipeline.get_lexing_pos pipeline pos in
+    let path =
+      match patho with
+      | Some p -> p
+      | None ->
+        let path = reconstruct_identifier pipeline pos None in
+        let path = Mreader_lexer.identifier_suffix path in
+        let path = List.map ~f:(fun {Location. txt; _} -> txt) path in
+        String.concat ~sep:"." path
+    in
+    if path = "" then `Invalid_context else 
+      let node = Mtyper.node_at typer pos in
+      let get_node_parent = Mbrowse.drop_leaf node in
+      let node_parent = Option.get get_node_parent in
+      let raw_node = snd (Mbrowse.leaf_node node) in
+      let node_name = Mbrowse.print_node () raw_node in 
```

Comment:
This printer is here for debugging purposes, I don't think we should rely on it for any program logic.

If you want to keep a "database", you could use a stronger type than `string` for the keys like a variant (or even a polymorphic variant).

But we might not need an intermediary database at all, unless we can think of another usage ?

Reply:
I agree. An intermediary database seems too much for our use case, especially as right now we don't have any other use for it. I'll use the earlier recommendation